src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -0,0 +1,131 @@
+import {RTC_ERRORS} from "../rtc_const";
+import "@citrix/ucsdk/CitrixWebRTC";
+
+export default class CitrixVDIStrategy extends CCPInitiationStrategyInterface {
+    constructor() {
+        super();
+        if (!isCitrixWebRTCSupported()) {
+            throw new Error('Citrix WebRTC redirection feature is NOT supported!');
+        }
+        window.getCitrixWebrtcRedir = function() {
+            const registryValue = Promise.resolve(1);
+            return new Promise(function(resolve, reject) {
+                //retrieve registry value internally
+                registryValue.then((v) => {
+                    resolve(v); }).catch(() => {
+                    reject(); })
+            })
+        }
+        window.CitrixWebRTC.initLog(global.connect.getLog());
+        console.log("CitrixVDIStrategy initialized");
+    }
+
+    // the following functions are rtc_peer_connection_factory related functions
+    // check if the browser supports early media connection
+    _isBrowserSupported(){
+        // Citrix WebRTC SDK doesn't support early media connection
+        return false;
+    }
+
+    _createRtcPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig) {
+        return new window.CitrixWebRTC.CitrixPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig);
+    }
+
+    // the following functions are rtc_session related functions
+    _gUM(constraints) {
+        return window.CitrixWebRTC.getUserMedia(constraints);
+    }
+
+    createOfferStateOnEnter(_pc, stream) {
+        stream.getTracks().forEach(track => {
+            _pc.addTransceiver(track, {});
+        });
+    }
+
+    acceptStateOnEnter(self, rtcSession) {
+        console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription`);
+
+        const answerSessionDescription = self._createSessionDescription({ type: 'answer', sdp: self._sdp });
+        console.log(`ConnectRTC - rtc_session - AcceptState - answerSessionDescription: `, answerSessionDescription);
+
+        rtcSession._pc.setRemoteDescription(answerSessionDescription, () => {
+            console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - success`);
+            var remoteCandidatePromises = Promise.all(self._candidates.map(function (candidate) {
+                var remoteCandidate = self._createRemoteCandidate(candidate);
+                self.logger.info('Adding remote candidate', remoteCandidate);
+                console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - Adding remote candidate - remoteCandidate`, remoteCandidate);
+                return rtcSession._pc.addIceCandidate(remoteCandidate);
+            }));
+            remoteCandidatePromises.catch(reason => {
+                console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - Adding remote candidate - failed`, reason);
```

Comment:
we usually only get CCP log for investigation, console log are not usually shared. but is the reason not using logger here is due to some limitations?

Reply:
As the CCP logger is not initialized  in this class, we can either initialize CCP logger in this class or remove these console logs.

Reply:
I've removed the console logs here